File: example/go/config/Ai/TriggerTick.go
package config.Ai
import (
    "config"
)
type TriggerTick interface
{
    internal static TriggerTick _create(Config.Stream os) {
        switch(os.ReadString()) {
            case "ConstValue":
                return config.Ai.Triggertick.ConstValue._create(os);
            case "ByLevel":
                return config.Ai.Triggertick.ByLevel._create(os);
            case "ByServerUpDay":
                return config.Ai.Triggertick.ByServerUpDay._create(os);
        }
        return null;
    }
}

File: example/java_ls/config/config/ai/TriggerTick.java
package config.ai;

public interface TriggerTick {
    static TriggerTick _create(configgen.genjava.ConfigInput input) {
        String tag = input.readStringInPool();
        switch (tag) {
            case "ConstValue":
                return config.ai.triggertick.ConstValue._create(input);
            case "ByLevel":
                return config.ai.triggertick.ByLevel._create(input);
            case "ByServerUpDay":
                return config.ai.triggertick.ByServerUpDay._create(input);
        }
        throw new IllegalArgumentException(tag + " not found");
    }
}
